File: ObserverV2Java/ObserverJava/src/src/Notifier.java
package src;

import include.INotifier;
import include.IObserver;
import include.ISubscriber;

import java.util.Iterator;
import java.util.List;

public class Notifier implements INotifier {

    Notifier(ISubscriber subs) {
        subscribers = subs;
    }

    public void _notify() {
        List<IObserver> observers = subscribers.getSubscribers();
        for (int i = 0; i < observers.size(); i++) {
            observers.get(i).update();
        }
    }

    private ISubscriber subscribers;
}

File: ObserverV3/src/Notifier.cpp
#include "../include/INotifier.h"
#include "../include/ISubscriber.h"
#include <iostream>

class Notifier: public INotifier{

    public:
        Notifier(ISubscriber* subs):subscribers(subs){
            std::cout << "Notifier::Notifier" << std::endl;
        }
        virtual ~Notifier(){
            std::cout << "Notifier::~Notifier" << std::endl;
        }

        void notify ()
        {
            std::list<IObserver*>::iterator it;
            std::list<IObserver*> observers = subscribers->getSubscribers();
            for (it = observers.begin() ; it != observers.end(); ++it)
            {
                (*it)->update();
            }
        }

    private:
        ISubscriber* subscribers;
};